app/src/main/java/com/andrew/tasky/agenda/presentation/screens/event_detail/EventDetailViewModel.kt

```diff
@@ -137,20 +133,13 @@ class EventDetailViewModel @Inject constructor(
                     }
                     is Resource.Success -> {
                         if (result.data != null) {
-                            if (result.data.doesUserExist) {
-                                if (!attendees.value.contains(result.data.attendee)) {
-                                    _attendees.value += result.data.attendee
-                                } else {
-                                    attendeeToastMessageChannel.send(
-                                        UiText.Resource(
-                                            resId = R.string.attendee_already_added
-                                        )
-                                    )
-                                }
+                            val attendee = result.data.copy(isGoing = true)
+                            if (!attendees.value.contains(attendee)) {
```

Comment:
It would probably make more sense here to check if there's already an attendee with the same ID rather than checking if there's an attendee that has exactly the same fields. With your implementation, if attendee A is marked as not going and it's already in the list, it would get added again here, since it would be marked as going so it's considered as a different attendee.
Instead you can do this:
```
val isAttendeePartOfEvent = attendees.value.any { it.id == result.data.id }
if(!isAttendeePartOfEvent) { ... }
```